stan/math/prim/arr/meta/is_vector.hpp

```diff
@@ -1,23 +1,45 @@
 #ifndef STAN_MATH_PRIM_ARR_META_IS_VECTOR_HPP
 #define STAN_MATH_PRIM_ARR_META_IS_VECTOR_HPP
 
+#include <stan/math/prim/arr/meta/is_vector.hpp>
 #include <stan/math/prim/scal/meta/is_vector.hpp>
 #include <vector>
-
+#include <type_traits>
 namespace stan {
 
-// FIXME: use boost::type_traits::remove_all_extents to
-//   extend to array/ptr types
+namespace internal {
+// FIXME(Steve): this is a stupid name
+template <typename T>
+using inherit_vector = std::vector<typename std::decay_t<T>::value_type,
+                                   typename std::decay_t<T>::allocator_type>;
+template <typename T>
+using is_same_vector
+    = std::is_same<typename std::decay_t<T>, internal::inherit_vector<T>>;
+}  // namespace internal
```

Comment:
This stuff is a bit wacky, the intent here is just to be able to verify that one vector is the same as another